### ISSUE TITLE
Compute and store whether an audit should be visible to the public

### ISSUE THREAD

**Issue by JeanMarie-TTS** [GSA-TTS/FAC#1537]
### Tasks

```[tasklist]
- [x] Add a boolean field is_public to SingleAuditChecklist in audit/models.py
- [x] Add code in the transition to SUBMITTED state to populate this field
```

- If entity_type == "tribal" set to False. Else set to True
- In a future enhancement we will add another check that will allow tribal entities to opt to have their submissions public.

**Comment by jadudm** [comment]
Copying this over from Slack, with some expansion:

Before too much implementation happens on this, I want to float this an idea. (and, I don't know where/when we prioritize this idea):

Should we, after cross-validation, have a process modeled on cross-val (e.g. a list of functions that take in the `sf_sac`) that add computed data to the model?

This `is_public` identifier is a *computed field*. There are <strike>probably</strike> others. (e.g. `cognizant_agency`). And, if the goal is to do it as a discrete step separate from ETL (e.g. a pre-ETL, "add computed fields" step), then we can/should do it in a way that is, perhaps, modeled on the cross-validation approach. (The architectural parallelism would make our lives easier, anyway. Cross-validation consumes the submission and returns zero or more errors. Once validated, the computed data pass consumes the submission and modifies the row in one or more ways.) 

Doing it this way suggests:

1. Submission happens.
2. The user triggers cross-validation.
3. The ability to move to certification becomes available.
4. A user triggers the move to certification, and we lock the submission.
5. The users certify, completing the audit.
6. We trigger computed data.
7. We trigger ETL.

This would guarantee that the computed step only happens once, and only once. 

To be clear: I don't know how/where we prioritize this. But we need to complete the submission process "as-is" before we add something like this. And I would like to see a thought-out design before we dive into implementation -- and I would like this prioritized after some other things that are still higher in the queue, unless we agree that it is essential to tackle this *now*.

**Comment by JeanMarie-TTS** [comment]
Discussed the topic at Co-work session. Here's what we decided to do for now:

1. Implement a method in audit/models.py that will determine and return a boolean value to indicate whther a report should be visible to the public
2. audit/etl.py will use this method to populate the relevant field in dissemination/models.py
3. Call the field is_report_public instead of is_public.


Deferred till later:

1. Should is_report_public be persisted in audit/models.py?
2. Should ETL be triggerred by a django view rather than by a state transisyion. This is to avoid the conditional import statement in the code today.
3. How should we handle other computed fields like cognizant_agency?

**Comment by pluribus-tony** [comment]
https://github.com/GSA-TTS/FAC/pull/1621
